trello-history/trello-history.md

```diff
@@ -0,0 +1,212 @@
+
+Keep in mind that power up capabilities are rendered in iframes. Yes. Please
+don't leave. The `renderHistory` and `askAuthorization` at the end of the
+promise flow are simple functions that both returns descriptive objects of how
+these iframes should be displayed. The important thing to understand is here:
+Each iframe targets a specific HTML page which runs its own JS script. That is,
+how can we possibily **share data** between those scripts ?
+
+See that `t` argument ? It's the power up toolbox, exposing helpful methods to
+interact with the context of the Trello board. For example, the `t.set()` method
+is to be used to store scalar values in the context (translations, for example,
+are a perfect use case for it). Once stored, a `t.get()` method can be called
+from any other script to get our translations back. Piece of cake !
+
+This can be missleading, though. Tricked by our daily habits, we spent some time
+trying to over decouple things (i.e get the history data from the API in the
+main power up script above, store it in the context and only use other scripts
+for templating). The Trello buffer has a limit size of 4096 characters, so no
+need to say we didn't have any chance to successfullly store our stringified
+history there ! We wandered towards locale storage and string compression to
+bypass that limitation but finally figured out this is not how it's was supposed
+to be done.
+
+One script per iframe / page to be shown, simple as that. Let's have a quick
+look on the history script:
+
+```js
+t.render(() => t.get('organization', 'shared', 'translations')
+  .then(renderHistory(t))
+)
+
+// renderHistory :: Object -> String -> _
+const renderHistory = t => translations => t.getRestApi().getToken()
+  .then(R.pipeP(
+    getCardHistory(t.getRestApi().appKey, t.getContext().card),
+    response => response.json(),
+    R.tap(() => document.getElementById('history').innerHTML = ''),
+    R.ifElse(
+      R.compose(R.gt(2), R.length),
+      R.tap(() => noHistory(translations)),
+      // drop the first element as it is exactly the same as the description
+      R.compose(R.map(renderCard), R.drop(1)),
+    ),
+  ))
+  .catch(error => openAuthorizeIframe(t))
+```
+
+The `t.render()` method ensures the history section is refreshed each time the
+user modifies the description of a card (so the previous description will be
+added to the history).
+
+Here you can see how we get the translations from the context. We're also using
+[ramda](https://ramdajs.com/docs/) to get full benefit of promise composition
+and fetch the card history from Trello. The `renderCard`  function does nothing
+more than transforming a `card` object returned from the API in a renderable
+HTML string, and that's all !
+
+![](https://github.com/jaljo/articles/raw/710f8e2d3561a5fa65bc6422b99aad88695871f3/trello-history/images/history.png)
+
+## Going further
+
+Of course, this problem could have been sloved in many another way, by
```

Comment:
I took some time to understand that the "problem" was the fact to not have the trello card history, maybe say something like: The problem that we don't have history description on trello card...